src/simplnx/DataStructure/BaseGroup.cpp

```diff
@@ -133,19 +135,21 @@ bool BaseGroup::isParentOf(const DataObject* dataObj) const
   return std::find_if(origDataPaths.begin(), origDataPaths.end(), [dataObj](const DataPath& path) { return dataObj->hasParent(path); }) != origDataPaths.end();
 }
 
-bool BaseGroup::insert(const std::weak_ptr<DataObject>& obj)
+Result<bool> BaseGroup::insert(const std::weak_ptr<DataObject>& obj)
 {
   auto ptr = obj.lock();
-  if(!canInsert(ptr.get()))
+  auto result = canInsert(ptr.get());
+  if(result.invalid())
   {
-    return false;
+    return result;
   }
   if(m_DataMap.insert(ptr))
   {
     ptr->addParent(this);
-    return true;
+    return {true};
```

Comment:
```suggestion
    return {{true}};
```
Since we use std::expected (from extraneous nonstd library) I believe we need the second bracket to construct it inside the Result object to pass it to the default constructor. This ensures that the valid() and invalid() Result wrapper methods function correctly. @JDuffeyBQ should probably verify I am correct in this behavior since I'm shaky on how the result is intended to function.